var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -108,23 +108,37 @@ class IntelParallelStudio(IntelInstaller):
 
     @property
     def blas_libs(self):
-        shared = True if '+shared' in self.spec else False
-        suffix = dso_suffix if '+shared' in self.spec else 'a'
-        mkl_integer = ['libmkl_intel_ilp64'] if '+ilp64' in self.spec else ['libmkl_intel_lp64']  # NOQA: ignore=E501
+        spec = self.spec
+        prefix = self.prefix
+        shared = '+shared' in spec
+
+        if '+ilp64' in spec:
+            mkl_integer = ['libmkl_intel_ilp64']
+        else:
+            mkl_integer = ['libmkl_intel_lp64']
+
         mkl_threading = ['libmkl_sequential']
-        if '+openmp' in self.spec:
-            mkl_threading = ['libmkl_intel_thread', 'libiomp5'] if '%intel' in self.spec else ['libmkl_gnu_thread']  # NOQA: ignore=E501
+
+        if '+openmp' in spec:
+            if '%intel' in spec:
+                mkl_threading = ['libmkl_intel_thread', 'libiomp5']
+            else:
+                mkl_threading = ['libmkl_gnu_thread']
+
         # TODO: TBB threading: ['libmkl_tbb_thread', 'libtbb', 'libstdc++']
+
         mkl_libs = find_libraries(
             mkl_integer + ['libmkl_core'] + mkl_threading,
-            root=join_path(self.prefix, 'mkl', 'lib', 'intel64'),
+            root=join_path(prefix, 'mkl', 'lib', 'intel64'),
+            shared=shared
+        )
+
+        # Intel MKL link line advisor recommends these system libraries
+        system_libs = find_system_libraries(
+            ['libpthread', 'libm', 'libdl'],
             shared=shared
         )
```

Comment:
thanks, LGTM. do you mind updating `intel-mkl` as well? (or if you have a clever idea of how to avoid bloody code duplication, feel free to fix it :smile: )


I will then test your patch tomorrow morning on Ubuntu 16.04+GCC with a whole lot of linear algebra packages (`petsc`, `superlu`, `trilinos`, etc). I expect your patch to work fine.